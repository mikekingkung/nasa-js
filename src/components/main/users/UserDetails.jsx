import axios from "axios";
import React, { useState, useEffect } from 'react';
import SearchUsersBar from './SearchUsersBar';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import UserLinks from './UserLinks';
import { ContactSupport } from "@material-ui/icons";

const UserDetails = () => {

  const usersUrl = `http://localhost:8080/users/getbyusername`;
    const userLinksUrl = `http://localhost:8080/users/getlinksbyusername`;

    const [userDetailsLoaded, setUserDetailsLoaded] = useState(false);
    const [userLinksIsLoaded, setUserLinksIsLoaded] = useState(false);

    const [userSearchText, setUserSearchText] = useState("Guest");
    const [userLinks, setUserLinks] = useState([]);

    const [userName, setUserName] = useState("");
    const [password, setPassword] = useState("");
    const [confirmPassword, setConfirmPassword] = useState("");
    const [title, setTitle] = useState("");
    const [firstName, setFirstName] = useState("");
    const [lastName, setLastName] = useState("");
    const [emailAddress, setEmailAddress] = useState("");
    const [phoneNumber, setPhoneNumber] = useState("");
    const [dob, setDob] = useState("");
    const [gender, setGender] = useState("");


    const userLinksSearch = "Guest";

    useEffect(() => {
     axios.get(`${usersUrl}/${userSearchText}`,
 
     {
          headers: {
            "Content-Type": "application/json",
            "Access-Control-Allow-Credentials": "*"
          }
        })
            .then(response => {
                const { data } = response;
                setUserName("Guest");
                setTitle(data.title);
                setFirstName(data.firstName);
                setLastName(data.lastName);
                setEmailAddress(data.emailAddress);
                setPhoneNumber(data.phoneNumber);
                setDob(data.dob);
                setGender(data.gender);
                setUserDetailsLoaded(true);
            }).catch(err => console.error("error getting user details", err));
      axios.get(`${userLinksUrl}/${userLinksSearch}`,
      {
          headers: {
            "Content-Type": "application/json",
            "Access-Control-Allow-Credentials": "*"
          }
        })
            .then(response => {
                const { data } = response;
                console.log("response from getUserLinks>>>>>>", response);
                console.log("data>>>>>>", data);               
                setUserLinks(data);
                console.log('userlinks>>>>>', userLinks);
                setUserLinksIsLoaded(true);
            }).catch(err => console.error("error getting userlinks", err));
      }, [userSearchText]); 

      
    //   const handleUserSearch = event => {
    //     event.preventDefault();

    //     console.log("key pressed:" , event.key)
    //     console.log(event.target.value);
    //     if (event.key === 'Enter'){
    //         console.log("setting search user text");
    //         setUserSearchText(event.target.value);
    //         setUserDetailsLoaded(false);
    //         setUserLinksIsLoaded(false);
    //     }
    // }

    console.log("userlinks before render", userLinks);

    // <div>
    // <SearchUsersBar
    //   searchUserText={userSearchText}
    //   handleUserSearch={handleUserSearch}
    // />


    return (
          <div>
          {userDetailsLoaded ?
            <Grid container spacing={6} >
               <Grid item xs={6} >
                <Typography/>
                <p>UserName:{userName}</p>
                <p>Title:{title}</p>
                <p>FirstName:{firstName}</p>               
                <p>LastName:{lastName}</p>
                <p>Email:{emailAddress}</p>
                <p>PhoneNumber:{phoneNumber}</p>
                <p>DOB:{dob}</p>
                <p>Gender:{gender}</p>
                <Typography/>
               </Grid>
            </Grid>
            :
            <p>Loading user details</p>
          }
          {userLinksIsLoaded ?
            <UserLinks links= {userLinks} />: <p>Loading user links</p>}
        </div>

    )
}

export default UserDetails;

