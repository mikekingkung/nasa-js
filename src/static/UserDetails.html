<!DOCTYPE html>
<html>
<head>
<style>
.nav {
  background-color: yellow; 
  list-style-type: none;
  text-align: center;
  margin: 0;
  padding: 0;
}

.nav li {
  display: inline-block;
  font-size: 20px;
  padding: 20px;
}
</style>
</head>
<body>

<ul class="nav">
    <li><a href="index.html">Landing</a></li>
    <li><a href="Discover.html">Discover</a></li>
    <li><a href="Topics.html">Topics</a></li>
    <li><a href="Links.html">Links</a></li>
    <li><a href="Secured.html">Secured</a></li>
    <li><a href="UserDetails.html">User Details</a></li>
    <li><a href="SignUp.html">Sign Up</a></li>
    <li><a href="image1.html">Image 1</a></li>
    <il><a href="image2.html">Image 2</a></li>
    <li><a href="image3.html">Image 3</a></li>
    <li><a href="About.html">About</a></li>
</ul>
<img src = "D:\Documents\react\nasa-js\src\images\bannerimage.png"> </img>
<p>User Details Content</p>
<p>UserName</p>
<p>Title</p>
<p>First Name</p>
<p>Last Name</p>
<p>Email</p>
<p>Phone Number</p>
<p>Date Of Birth</p>
<p>Gender</p>
<p>Saved Links</p>
<p>Link1</p>
<p>Link2</p>
<p>Link3</p>

<img src = "D:\Documents\react\nasa-js\src\images\footerimage.png"> </img>
</body>
</html>

