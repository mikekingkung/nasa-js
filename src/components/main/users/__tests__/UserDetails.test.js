import React from 'react';
import { render} from '@testing-library/react';
import UserDetails from '../UserDetails';

describe('UserDetails Page component', () => {
    it('Correctly renders' ,() => {
        const wrapper = render(<UserDetails/>);
        expect(wrapper).toMatchSnapshot();
    });
});