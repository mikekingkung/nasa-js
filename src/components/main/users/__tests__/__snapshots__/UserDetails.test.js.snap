// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails Page component Correctly renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <label>
            Search:
          </label>
          <input
            id="search"
            placeholder=""
            type="search"
          />
        </div>
        <br />
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            />
            <p>
              UserName:
              
            </p>
            <p>
              Title:
              
            </p>
            <p>
              FirstName:
              
            </p>
            <p>
              LastName:
              
            </p>
            <p>
              Email:
              
            </p>
            <p>
              PhoneNumber:
              
            </p>
            <p>
              DOB:
              
            </p>
            <p>
              Gender:
              
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <label>
          Search:
        </label>
        <input
          id="search"
          placeholder=""
          type="search"
        />
      </div>
      <br />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          />
          <p>
            UserName:
            
          </p>
          <p>
            Title:
            
          </p>
          <p>
            FirstName:
            
          </p>
          <p>
            LastName:
            
          </p>
          <p>
            Email:
            
          </p>
          <p>
            PhoneNumber:
            
          </p>
          <p>
            DOB:
            
          </p>
          <p>
            Gender:
            
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
